libqtile/config.py

```diff
@@ -832,7 +832,82 @@ def convert_deprecated_list(vals: list[str], name: str) -> re.Pattern:
     return re.compile(regex_input)
 
 
-class Match:
+class _Match:
+    """Base class to implement bitwise logic methods for Match objects."""
+
+    def compare(self, client: base.Window) -> bool:
+        return True
+
+    def __invert__(self) -> _InvertMatch:
+        return _InvertMatch(self)
+
+    def __and__(self, other: _Match) -> _MatchAllOf:
+        if not isinstance(other, _Match):
+            raise TypeError
+
+        return _MatchAllOf(self, other)
+
+    def __or__(self, other: _Match) -> _MatchOneOf:
+        if not isinstance(other, _Match):
+            raise TypeError
+
+        return _MatchOneOf(self, other)
+
+    def __xor__(self, other: _Match) -> _MatchOnlyOneOf:
+        if not isinstance(other, _Match):
+            raise TypeError
+
+        return _MatchOnlyOneOf(self, other)
+
+
+class _InvertMatch(_Match):
+    """Wrapper to invert the result of the comparison."""
+
+    def __init__(self, match: _Match):
+        self.match = match
+
+    def compare(self, client: base.Window) -> bool:
+        return not self.match.compare(client)
+
+    def __repr__(self) -> str:
+        return "<_InvertMatch(%r)>" % self.match
```

Comment:
you don't need the `<>` here right?

Reply:
No. I've always done that out of habit if nothing else!

I was also thinking that we could make those wrapper classes public in case people wanted to use them instead of the operators.

Reply:
Yep, agreed on making them public. If we drop the `<>`, I think you can copy paste them into a python interpreter, which is my understanding of what repr() is supposed to do. With `<>`, you'd have to manually delete them all.

Reply:
All the other `repr`s in that file have the `<>`!

Reply:
Ah, reading the actual docs, it's either or: https://docs.python.org/3/library/functions.html#repr

my read is: if you can't copy and paste it into eval() it should have <>, but otherwise it should not.